### ISSUE TITLE
Markdown Editing: Spaces in Table skip 4 characters

### ISSUE THREAD

**Issue by RickStrahl** [ajaxorg/ace#4419]
When using arrow keys to navigate through a Pipe table in Markdown mode, the cursor will on various characters jump inconsistently 4 characters forward instead of moving the expected 1 character at a time. 

Try the following example table in the [Kitchen Sink](https://ace.c9.io/kitchen-sink.html?doc=Markdown) using `Markdown` syntax:

```markdown
| Project                      | Mon | Tue | Wed | Thu | Fri |
|:-----------------------------|:---:|:---:|:---:|:---:|:---:|
|                              |     |     |     |     |     |
|                              |     |     |     |     |     |
|                              |     |     |     |     |     |
```

To make it clear that the text is just spaces in between the table lines show invisibles.

Now move through the columns of the table using the right arrow key.

Observed: 

At first it works as expected, but as you get to the middle of the first column the cursor skips 4 characters. In the next column it immediately jumps to the end of the column (4 spaces again). In the next column it just goes normal.

In short, there's something interfering inside of the pipe table syntax that's causing the cursor to jump 4 characters on some single arrow operations. I can't really make out a pattern when the skips occur, so it appears semi-random except it appears to be consistently repeatable for the given table format.

**Comment by nightwing** [comment]
Looks like this was done [intentionally](https://github.com/ajaxorg/ace/blob/e9f53ddf3c789f0042d83a0c9c3163021d028159/lib/ace/selection.js#L424), to make indentation with spaces behave like indentation with tabs, (textmate editor behaves like this too).
I am not sure if we should remove this feature, or limit it only to the beginning of the line, but until this is changed, you can use `editor.selection.wouldMoveIntoSoftTab=()=>false` as a temporary workaround

**Comment by RickStrahl** [comment]
@nightwing Yes - the general editor Soft Tab skipping of  tabs with arrow navigation is totally appropriate and definitely shouldn't be turned off.

But... in the case of the Markdown PipeTable there seems to be something additional happening as the actual text doesn't contain any tabs - only spaces. Yet it looks like soft tab skipping occurs inside of the pipe table. Using your workaround and disabling the auto softtab behavior the table works correctly, but unfortunately that disables the behavior everywhere in the editor which is not the desired behavior.

The way it works right now in the pipe table feels very erratic in behavior.  Using arrow keys to move over a row, in the first column I get the 4 char skip after 3 characters, on the second column after 1st, third column not at all, last column on 1st, forth not at all, fifth on first (every other one?). 

I guess I'm not clear on what's actually happening or even what is supposed to be happening.
